### ISSUE TITLE
Support of Connectivity Kit

### ISSUE THREAD

**Issue by ostocker** [Somfy-Developer/Somfy-TaHoma-Developer-Mode#12]
Do you also support the Connectivity Kit?
https://www.somfy.de/produkte/1870755/connectivity-kit

**Comment by llavorel-somfy** [comment]
The Connectivity Kit does not support the Developer Mode

**Comment by BillGOH** [comment]
Support for Connectivity Kit would be highly appreciated. Local mode works with Apple Home Kit, so why not with Somfy local API?

**Comment by FliegePit** [comment]
Hi all,
I just recently had to switch my old Connexoon device since it was just dead and did not work anymore. Now I have to find out, that the Connectivity Kit comes with a reduced App Feature Set in the Tahoma App - especially no way to automate events :(.
Since I don't want to go through the whole procedure of buying a third device (and thus having to reset all devices again and learning them in again ...) I would like to program my own logic.
In general: all I want is to turn down shutters in the evening at sunset ... not a big deal, but disabled.

Long story short: please Activate developer Mode for Connectivity Kit?!

Or: enable event features in Tahoma App for Connectivity Kit? I am pretty sure, that these events are defined on the server side and not within the connectivty kit ...

**Comment by B0rax** [comment]
Any news on this? Will the connectivity kit be compatible with the local api? This is a huge drawback and it is not documented anywhere. Please at least put it in the description on your website. Local api would help immensely with integration in house automation.

**Comment by 9Mad-Max5** [comment]
With Somfy it is always an adventure what products are capable of what and how.
I mean why is it not possible to have the local IP on this device?
As many already sad, what is the difference to have it working with HomeKit?

**Comment by Tamas-Toth-ebola** [comment]
**Please** Somfy!

Add local API support to Connectivity Kit also as the local only working method is the most secure way to use any home automation devices which/therefore it should be also one of the most important goals of the manufacturers also, at least to nowadays. **I understand that cloud connection could be business advantage but there are other aspects which should be condsiderable, at least with compromises. I think a discrete/hidden developer mode as local API support could serve both needs _(as it does on your other devices)_.**

@llavorel-somfy :point_up:

**Comment by corneliusmunz** [comment]
Hi Somfy, @llavorel-somfy  I would highly appreciate the support of the local API support for the Connectivity Kit

**Comment by ninut** [comment]
Hi Somfy, @llavorel-somfy to add another point to @Tamas-Toth-ebola excellent write up: the device is called connectivity kit. The name implied and the product page is not informing the customer, that you can't use this particular box to connect (!) it to any third party apps. 
Nothing, neither the product page, nor the information page regarding the developer mode hints which devices were compatible with the local API. 
So I am asking is the devices on a hardware level not compatible, is it able to support the dev mode but not enables yet (next question is "when will it be") or should I send it back and you're updating the named page's to avoid further ewaste?

**Comment by Tamas-Toth-ebola** [comment]
@ninut hi!

I think I 'know' the answer to your question, but of course, this is just my subjective opinion:

- Local API is a relatively new solution/function/possibility of the Somfy - Tahoma box. If I'm right it had announced 2022, and just terminated 'its' older version in the summer _(https://developer.somfy.com/ | https://github.com/home-assistant/core/issues/69558#issue-1195803530)_. The open-source local API project has just 2 PR altogether, so it seems fresh. If we check the happenings in the repo, we can see some problems/requests already, so maybe Somfy does not really feel like a final/public solution yet.
- The 'Connection Kit' limitation _(baically/usually)_ can not be HW based. The device itself is nothing more than a multi-radio controller with the corresponding software to manage them. The software is software, which could be anything else than the current one. The only limitation could be if the software is not refreshable, but nowadays a simple wall socket also could be firmware updated so I do not really think, that 'Connectivity Kit' can not allow it _(especially as we know that the device can update itself, and it already scans for new firmware version automatically)_. It also has a WiFi radio, so it could be reachable through LAN instead of the internet. So in the end with a new software, I'm pretty sure that we could reach its local API in theory.
- The problem that the 'new' software for it is time/cost and this can hurt Somfy, meanwhile, the main experiment on the Tahoma box is not matured yet. So I have to say, that local API is a mandatory thing on any nowadays available _(known brand offered)_ smart-home products, as the offered security is more than basic, but I don't know yet how Somfy feel this and if agrees with me, when will be the point, if the software on the Tahoma box considered as final, to allow to 'fork' it to other devices. Until it, the problems on more devices could escalate a lot far than only with one device.
- Because 'local API' is a relatively new solution, Somfy did not have time/capacity to update its documentation yet, especially if the solution behind is not even considered full-final yet.

**So I think what we can do is just wait, nothing more, and hope for the best.**

But this is just my opinion.

**Comment by demichve** [comment]
Fully agree! I am currently struggling with Cloud API using Connectivity Kit.
If possible, Somfy should enable the activation of local API.

**Comment by BillGOH** [comment]
It's a shame that Somfy does not communicate in this matter. Somfy cloud connection with Connectivity Kit ist not reliable. I for myself have decided not to use Somfy products any more - and I cannot recommend them.

**Comment by Tamas-Toth-ebola** [comment]
Just some bonus words about the mentioned security background by me.

The Tahoma mobile app, which is responsible for the configuration of the Connectivity Kit expects to configure the corresponding network through a public WiFi connection, so no hidden network can work with it. Is it really valid/acceptable nowadays - from a security aspect - to have a public WiFi network? It is horrible. To use the Connectivity Kit I had to define a special guest network - outside of my 'real internal' LAN - to accept 'its requirements'. Without my  double router based network chain, basically, I should have to expose my 'internal router' _(which is responsible for my whole home network)_ with its guest network config to the publicity. 

I have to say again. What could be more important in the smart-home segment than security?

But as I wrote in my first line, this is just an 'addendum', nothing more.

**Comment by llavorel-somfy** [comment]
Unfortunately, Connectivity kit doesn't share the same technical architecture than our Tahoma range. 
After careful evaluation, we have concluded that the current memory capacity of our Connectivity Kit is not sufficient to support the demands of the Developer Mode without sacrificing other product features.

**Comment by B0rax** [comment]
Some people may not need all the functions that the connectivity kit has to offer, but would really benefit from the developer mode. Maybe it would be possible to have a “developer version” which does have more limited functionality in exchange for the developer mode?

**Comment by BillGOH** [comment]
Sorry @llavorel-somfy , that does not make sense:
1. Connectivity Kit supports Apple Homekit, which works locally.
2. Somfy has pledged to support Matter, which also works locally.
So, as I wrote before, Somfy is not as trustworthy partner. No Somfy products for me anymore.

**Comment by grafdezimal** [comment]
Hi, @llavorel-somfy 
just learned from BernoldNL in #181 that Connectivity Kit does not support Developer Mode and you are not planning to add it because of architecture differences. Ok, so far understood. Is there any other possibility to control my new somfy devices from third party tool? I have 2 motors I need to control, even connectivity kit seems a bit overkill for this job, do I really need to buy a TaHoma Switch now?
Is there any other "simpler" API I could activate/use? Cloud or local?

**Comment by llavorel-somfy** [comment]
Hi @grafdezimal 

Unfortunately, the developerMode is the only official service we provide for integration with third party platforms that we do not develop at Somfy.
